lib/api/v3/queries/queries_api.rb

```diff
@@ -41,14 +41,27 @@ class QueriesAPI < ::API::OpenProjectAPI
             before do
               @query = Query.find(params[:id])
               @representer = QueryRepresenter.new(@query)
+              authorize_as_not_found(:show)
```

Comment:
You are right... I slightly modified `authorize_by_policy`, it will now behave like `authorize` methods from `root.rb` and accept a block:

````ruby
authorize_by_policy(:show) do
  raise API::Errors::NotFound
end
````